apps/system/build/build.js

```diff
@@ -99,10 +100,29 @@ SystemAppBuilder.prototype.integrateLockScreenInputpad = function(options) {
   utils.writeContent(systemIndexFile, replacedIndexContent);
 };
 
+SystemAppBuilder.prototype.writeSearchPartnerJson = function() {
+
+  var variantFile = utils.getFile(this.distDirPath, 'variant.json');
+  var searchProviderJson =
+    utils.getJSON(utils.getFile(this.gaiaDirPath, 'shared', 'js',
+                                'search_providers_input.json'));
+
+  if (variantFile.exists()) {
+    var variantJson = utils.getJSON(variantFile);
+    if ('search_partner_code' in variantJson) {
+      searchProviderJson.partner_code = variantJson.search_partner_code;
+    }
+  }
+  var outFile = utils.getFile(this.gaiaDirPath, 'shared', 'js',
+                              'search_providers.json');
```

Comment:
Shared resource would be copied into each app's stage folder before its own build, so the modification should not work.

If you want every app to use modified search_providers.json, we should modify it before webapp-shared (see pre-app.js), perhaps we can make a new build module or combine it into contacts-import-service.js and rename it because they are all trying to modify shared resource.